pkg/provider/podsTracker.go

```diff
@@ -148,6 +155,23 @@ func (pt *PodsTracker) processPodUpdates(ctx context.Context, pod *v1.Pod) bool
 	ctx, span := trace.StartSpan(ctx, "PodsTracker.processPodUpdates")
 	defer span.End()
 
+	lastEventCheck := pt.lastEventCheck
```

Comment:
Please add a unit test for the added functionality.

Reply:
done